# Using Different Models with Autogen

## OpenAI - Already seen

# Ollama - Local hosted

In [3]:
from autogen_core.models import UserMessage
from autogen_ext.models.ollama import OllamaChatCompletionClient

# Assuming your Ollama server is running locally on port 11434.
ollama_model_client = OllamaChatCompletionClient(model="llama3.2")

response = await ollama_model_client.create([UserMessage(content="Who are you, who created you?", source="user")])
print(response)
await ollama_model_client.close()


finish_reason='stop' content='I\'m an artificial intelligence model known as Llama. Llama stands for "Large Language Model Meta AI."' usage=RequestUsage(prompt_tokens=33, completion_tokens=23) cached=False logprobs=None thought=None


finish_reason='stop' content='The capital of France is Paris.' usage=RequestUsage(prompt_tokens=32, completion_tokens=8) cached=False logprobs=None thought=None

finish_reason='stop' content="I am a computer program designed to simulate conversation and answer questions to the best of my ability. I'm a type of artificial intelligence (AI) called a large language model, which means I've been trained on a massive dataset of text from various sources.\n\nMy primary function is to assist users with information and tasks, and I do this by processing natural language inputs and generating human-like responses. I don't have personal experiences, emotions, or consciousness like humans do, but I'm designed to be helpful and informative.\n\nI was created by a team of researcher-engineers at Meta AI, which is a subsidiary of Meta Platforms, Inc. The development of my technology involved extensive research in natural language processing (NLP), machine learning, and cognitive architectures.\n\nMy training data consists of a massive corpus of text from various sources, including books, articles, research papers, and websites. This corpus was used to fine-tune my understanding of language patterns, semantics, and context.\n\nI'm constantly learning and improving through interactions with users like you, so I appreciate any feedback or corrections that can help me become a better conversational AI!" usage=RequestUsage(prompt_tokens=33, completion_tokens=226) cached=False logprobs=None thought=None

In [ ]:
from autogen_core.models import UserMessage
from autogen_ext.models.ollama import OllamaChatCompletionClient

# Assuming your Ollama server is running locally on port 11434.
ollama_model_client = OllamaChatCompletionClient(model="llama3.2")

response = await ollama_model_client.create([UserMessage(content="Who are you, who created you?", source="user")])
print(response)


finish_reason='stop' content='I am an artificial intelligence model known as Llama. Llama stands for "Large Language Model Meta AI."' usage=RequestUsage(prompt_tokens=33, completion_tokens=23) cached=False logprobs=None thought=None


In [6]:
from autogen_agentchat.agents import AssistantAgent
agent_1 = AssistantAgent(name='MyOllamaAgent',model_client=ollama_model_client)

In [7]:
result = await agent_1. run(task='Tell me something about you?')

In [8]:
result

TaskResult(messages=[TextMessage(id='a6b436b1-ca2a-430f-bf42-cb46e2ac0185', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 5, 5, 19, 6, 598002, tzinfo=datetime.timezone.utc), content='Tell me something about you?', type='TextMessage'), TextMessage(id='ad293544-9fd3-4610-9826-536bf9855c2c', source='MyOllamaAgent', models_usage=RequestUsage(prompt_tokens=55, completion_tokens=136), metadata={}, created_at=datetime.datetime(2025, 7, 5, 5, 19, 16, 631317, tzinfo=datetime.timezone.utc), content="I'm an artificial intelligence designed to assist and communicate with users like you. I have been trained on a vast amount of text data, which enables me to understand and respond to a wide range of questions and topics.\n\nMy primary function is to provide helpful and accurate information, answer questions, and engage in conversations to the best of my abilities. I can process natural language inputs and generate human-like responses.\n\nI don't have personal 

TaskResult(messages=[TextMessage(id='a6b436b1-ca2a-430f-bf42-cb46e2ac0185', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 5, 5, 19, 6, 598002, tzinfo=datetime.timezone.utc), content='Tell me something about you?', type='TextMessage'), TextMessage(id='ad293544-9fd3-4610-9826-536bf9855c2c', source='MyOllamaAgent', models_usage=RequestUsage(prompt_tokens=55, completion_tokens=136), metadata={}, created_at=datetime.datetime(2025, 7, 5, 5, 19, 16, 631317, tzinfo=datetime.timezone.utc), content="I'm an artificial intelligence designed to assist and communicate with users like you. I have been trained on a vast amount of text data, which enables me to understand and respond to a wide range of questions and topics.\n\nMy primary function is to provide helpful and accurate information, answer questions, and engage in conversations to the best of my abilities. I can process natural language inputs and generate human-like responses.\n\nI don't have personal experiences, emotions, or opinions like humans do, but I'm designed to be neutral and informative. My goal is to help users find the information they need, learn new things, or simply have a conversation.\n\nWhat would you like to talk about?", type='TextMessage')], stop_reason=None)

# GEMINI

In [11]:
from autogen_core.models import UserMessage
from autogen_ext.models.openai import OpenAIChatCompletionClient

model_client = OpenAIChatCompletionClient(
    model="gemini-1.5-flash-8b",
    api_key="AIzaSyDrXnfYcy8hUz9QKKdAmpVf1u--vU8UThc",
)

response = await model_client.create([UserMessage(content="What is the capital of India?", source="user")])
print(response)
await model_client.close()


finish_reason='stop' content='New Delhi\n' usage=RequestUsage(prompt_tokens=7, completion_tokens=3) cached=False logprobs=None thought=None


finish_reason='stop' content='Paris\n' usage=RequestUsage(prompt_tokens=7, completion_tokens=2) cached=False logprobs=None thought=None

# Open Router

In [14]:
open_router_api_key = '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'

open_router_model_client =  OpenAIChatCompletionClient(
    base_url="https://openrouter.ai/api/v1",
    model="openrouter/cypher-alpha:free",
    api_key = open_router_api_key,
    model_info={
        "family":'deepseek',
        "vision" :True,
        "function_calling":True,
        "json_output": False
    }
)


response = await open_router_model_client.create([UserMessage(content="What is the capital of France?", source="user")])

print(response)

finish_reason='stop' content="The capital of France is Paris. It's known for its historic landmarks like the Eiffel Tower, Louvre Museum, and Notre-Dame Cathedral. The city attracts millions of tourists each year with its rich culture, art, and cuisine. Let me know if you need more information about France or any other European country!" usage=RequestUsage(prompt_tokens=392, completion_tokens=63) cached=False logprobs=None thought=None


finish_reason='stop' content='The capital of France is **Paris**. Known for iconic landmarks such as the Eiffel Tower, the Louvre Museum, and Notre-Dame Cathedral, Paris has long served as the political, cultural, and economic heart of France. It houses key government institutions, including the official residence of the French President (Élysée Palace) and the National Assembly.' usage=RequestUsage(prompt_tokens=13, completion_tokens=409) cached=False logprobs=None thought=None